bitcoin/src/blockdata/script.rs

```diff
@@ -1222,6 +1236,66 @@ mod test {
         script = script.push_opcode(OP_CHECKSIG); comp.push(0xACu8); assert_eq!(&script[..], &comp[..]);
     }
 
+    #[test]
+    fn test_p2pk_pubkey_bytes() {
+        // happy path
+        let key_str = "0411db93e1dcdb8a016b49840f8c53bc1eb68a382e97b1482ecad7b148a6909a5cb2e0eaddfb84ccf9744464f82e160bfa9b8b64f9d4c03f999b8643f656b412a3";
+        let key = PublicKey::from_str(key_str).unwrap();
+        let p2pk = Script::builder().push_key(&key).push_opcode(OP_CHECKSIG).into_script();
+        let actual = p2pk.p2pk_pubkey_bytes().unwrap();
+        assert_eq!(actual.to_vec(), key.to_bytes());
```

Comment:
Nit: `to_vec()` shouldn't be needed.

Reply:
[b14580e](https://github.com/rust-bitcoin/rust-bitcoin/pull/1412/commits/b14580e440fcad7413ba59728de8c83e00f5986e) failed because of one of the feature sets. to_vec() fixed it across the board

Reply:
Out of curiosity, what was the error message, please? `PartialEq` between those type [is implemented](https://doc.rust-lang.org/std/cmp/trait.PartialEq.html#impl-PartialEq%3C%26%5BU%5D%3E-for-Vec%3CT%2C%20A%3E) so I can't wrap my head around why it'd fail.

Reply:
Sure, if you look in the commit and click the X notating that the workflows have failed, you can see the one workflow that failed and will see this message: 

```
error[E0277]: can't compare `&[u8]` with `std::vec::Vec<u8>`
    --> bitcoin/src/blockdata/script.rs:1246:9
     |
1246 |         assert_eq!(actual, key.to_bytes());
     |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ no implementation for `&[u8] == std::vec::Vec<u8>`
     |
     = help: the trait `std::cmp::PartialEq<std::vec::Vec<u8>>` is not implemented for `&[u8]`
     = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)

error[E0277]: can't compare `&[u8]` with `std::vec::Vec<u8>`
    --> bitcoin/src/blockdata/script.rs:1266:9
     |
1266 |         assert_eq!(actual, key.to_bytes());
     |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ no implementation for `&[u8] == std::vec::Vec<u8>`
     |
     = help: the trait `std::cmp::PartialEq<std::vec::Vec<u8>>` is not implemented for `&[u8]`
     = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0277`.
error: could not compile `bitcoin`.
```

Reply:
This is the command that was ran `cargo test --verbose --no-default-features --features=std`

Reply:
FWIW It fails on Rust 1.41.1 not stable.

Reply:
@tcharding that explains it, thanks! Impl was added in 1.46

@connormullett adding `*` in front of both arguments should've worked but let it be. No need to micro-perf-optimize tests like this.